experimental/Schemes/DerivedPushforward.jl

```diff
@@ -75,7 +75,7 @@ end
 =#
 
 function rank(phi::FreeModuleHom{FreeMod{T}, FreeMod{T}, Nothing}) where {T<:FieldElem}
-  return ngens(domain(phi)) - left_kernel(sparse_matrix(phi))[1]
+  return ngens(domain(phi)) - nrows(kernel(sparse_matrix(phi), side = :left))
```

Comment:
I think we need a function nullity - to return the rank of the kernel only...

Reply:
I add it to the list of future improvements.